### ISSUE TITLE
[0.17.0] LOW_MEM + social_network -> huge shm-file consumption

### ISSUE THREAD

**Issue by bitfag** [GolosChain/golos#478]
0.17.0 LOW_MEM node with the following plugins consumes shm-file as much as 0.16.4 (~10 GB):

0.17.0:
```
plugin = chain p2p json_rpc webserver network_broadcast_api witness database_api private_message market_history
```

0.16.4:
```
enable-plugin = witness market_history
```

But **adding `social_network` plugin increases a shm-file consumption dramatically**: 12G+ at 80% sync progress.

I reviewed differences between `database_api.cpp` from 0.16.4 and `social_network.cpp` from 0.17.0 and found IS_LOW_MEM defines was moved correctly. So it's not yet clear to me where such increased consumption comes from.

**Comment by bitfag** [comment]
OK seems like this is happening because `tags` plugin is now part of the `social_network`, thus `tag_visitor` creates additional databases:

```cpp
                    const auto &tag_obj = _db.create<tag_object>([&](tag_object &obj) {
                        obj.name = tag;
                        obj.comment = comment.id;
                        obj.parent = parent;
                        obj.created = comment.created;
                        obj.active = comment.active;
                        obj.cashout = comment.cashout_time;
                        obj.net_votes = comment.net_votes;
                        obj.children = comment.children;
                        obj.net_rshares = comment.net_rshares.value;
                        obj.children_rshares2 = comment.children_rshares2;
                        obj.author = author;
                        obj.hot = hot;
                        obj.trending = trending;
                    });
                    add_stats(tag_obj, get_stats(tag));
```

**Comment by bitfag** [comment]
Would be nice to have a switch to disable tag_visitor. Locally I'll disable it via `#ifndef IS_LOW_MEM`.

**Comment by afalaleev** [comment]
We can add a parameter for disable it via config-file. But for me, your decision with disabling via IS_LOW_MEM seems more intuitively clear. Do you want to create pull request to golos (branch golos-v0.17.0)?

**Comment by bitfag** [comment]
OK, created a PR, but it looks like there is something else.

Golosd 0.17 **without** `social_network` consumes 9.9G exactly as 0.16.4. According to my expectation it should consume 9.9G **with** `social_network`.

E.g., the following configurations should give same results:

```
plugin = chain p2p json_rpc webserver network_broadcast_api witness database_api market_history social_network
```

```
enable-plugin = witness market_history
public-api = database_api login_api market_history_ap
```

Am I missing something?

**Comment by afalaleev** [comment]
Fixed in #580
